### ISSUE TITLE
Quick failure diagnosis

### ISSUE THREAD

**Issue by ruchim** [DataBiosphere/job-manager#349]
The operations team wants the quickest possible way to diagnose why failures have occurred. fewest clicks possible. When a task has failed, there is information in the failure message in the metadata that could be useful for the user. In addition, if the workflow has a lot of tasks then it may be annoying to scroll through them and find the one(s) that failed. When a job has failed we want troubleshooting the failure to be as easy as possible. I think a quick step we could take towards that would be to surface this failure information on the details page.

AC:
The operations team feels they can more quickly and easily diagnose failures via an error panel on the details page. 
This panel displays the first five failures that occurred with quick links to their logs and execution dirs. When on the parent workflow page, this should include if tasks failed inside of subworkflows, or inside of scatters each shard should show in this list. 

Out of Scope:
They have requested that when a job has failed, the links to the error logs of the task that failed first be on the list view. This has been included in ticket #204 
In future designs we have a drop down for the error panel and it could potentially have more failures than 5 listed.

**Comment by rsasch** [comment]
@ruchim To confirm, this is referring to this part of the wireframe:

<img width="1066" alt="error-panel" src="https://user-images.githubusercontent.com/1713505/43209067-c796f92a-8ff9-11e8-8f7a-38c43fdfd866.png">

This display is limited to 3 failed tasks or shards. Will that actually make sense for a scattered task?

**Comment by rsasch** [comment]
I'm also unclear on how we are going to access the metadata for sub-workflows; we currently only get data on one task/job (via ID) at a time.

**Comment by rsasch** [comment]
**Blocked**: waiting on confirmation of UI and what to do in the case of sub-workflows and scattered tasks.

**Comment by kbergin** [comment]
Yes, that portion of the UI is what the ticket refers to.

Scattered tasks - yes show the shards that have failed. If it's possible to expand the panel to see more failures like you showed this morning that'd be good, but once a user sees past like 10 failures in the same scatter they're going to be going to manual investigation either way

I think you are correct that we can't currently show tasks from subworkflows in that panel. In the (near) future the intention is for subworkflows to all be on the same details page, so they would then be available. Not sure if that influences how it is implemented. Feel free to iterate in the way that makes the most sense based on this future plan.

**Comment by rsasch** [comment]
@kbergin Thanks for the info.  I'll see what kind of flexibility I have with the implementation of "cards" in the UI (there doesn't seem to be an easy native way to make them expand/contract, unlike the "panels" that list tasks...etc. and are collapsible).

**Comment by rsasch** [comment]
closed in favor of #451
